### PR TITLE
Feature/new block

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Adds the new Oak shingle block with some abstractions to easily add more shingles
- Reworks all custom minecolonies models to look nice in itemFrames, the inventory and the hand.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java [hunk 1]
```diff
@@ -26,6 +26,9 @@
 import com.minecolonies.coremod.tileentities.TileEntityInfoPoster;
 import com.minecolonies.structures.event.RenderEventHandler;
 import com.minecolonies.structures.helpers.Settings;
+import net.minecraft.block.Block;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java]: MINOR Remove this unused import 'net.minecraft.block.Block'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java [hunk 2]
```diff
@@ -26,6 +26,9 @@
...
+import net.minecraft.block.BlockPlanks;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java]: MINOR Remove this unused import 'net.minecraft.block.BlockPlanks'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java [hunk 3]
```diff
@@ -26,6 +26,9 @@
...
+import net.minecraft.block.SoundType;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java]: MINOR Remove this unused import 'net.minecraft.block.SoundType'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 44 issues
* ![MINOR][MINOR] 43 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [ModBlocks.java#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java#): Add a new line at the end of this file. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00113)
1. ![MINOR][MINOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [init#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [management#](https://github.com/Minecolonies/minecolonies/blob/bbce03a026d9d23004f3536dd45c0dd47a1c434e/src/main/java/com/minecolonies/coremod/colony/requestsystem/management#): Add a 'package-info.java' file to document the 'management' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 41 issues
* ![MINOR][MINOR] 40 minor
* ![INFO][INFO] 1 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [ModBlocks.java#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java#): Add a new line at the end of this file. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00113)
1. ![MINOR][MINOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [init#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [management#](https://github.com/Minecolonies/minecolonies/blob/44fb7da18a720fbb4c34abc79a8e2618c5a3d088/src/main/java/com/minecolonies/coremod/colony/requestsystem/management#): Add a 'package-info.java' file to document the 'management' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MineShadow7** [comment]
Screenshots? (I'm not a reviewer but still)

**Pull request merged**